File: src/main/java/com/twalthr/flink/examples/Example_01_DataStream_Motivation.java
package com.twalthr.flink.examples;

import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.util.CloseableIterator;

/** Basic example of generating data and printing it. */
public class Example_01_DataStream_Motivation {

  public static void main(String[] args) throws Exception {
    StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

    CloseableIterator<Customer> it =   env.fromElements(ExampleData.CUSTOMERS)
        .executeAndCollect();

        it.forEachRemaining(System.out::println);
  }
}

File: src/main/scala/ch/iwerk/flink/examples/twalthr/Example_01_DataStream_Motivation.scala
package ch.iwerk.flink.examples.twalthr

import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment

object Example_01_DataStream_Motivation {

  def main(args: Array[String]): Unit = {
    val env = StreamExecutionEnvironment.getExecutionEnvironment

    env.fromCollection(ExampleData.CUSTOMERS)
      .executeAndCollect()
      .forEachRemaining(println)


     env.fromElements(ExampleData.CUSTOMERSArray: _*)
        .executeAndCollect
        .forEachRemaining(println)
  }




}